python/lib/scheduler/dmod/scheduler/resources/resource_manager.py

```diff
@@ -170,126 +184,159 @@ def allocate_single_node(self, cpus: int, memory: int) -> List[Optional[Resource
         #Fit the entire allocation on a single resource
         self.validate_allocation_parameters(cpus, memory)
 
-        for res in self.get_useable_resources():
-            if res.cpu_count >= cpus and res.memory >= memory:
-                allocations = []
-                mem = int(memory / cpus)
-                for i in range(cpus):
-                    alloc = self.allocate_resource(resource_id=res.resource_id, requested_cpus=1, requested_memory=mem)
-                    if alloc is None:
-                        allocations = None
-                        break
-                    else:
-                        allocations.append(alloc)
-                if allocations is not None:
-                    return allocations
-        return [None]
+        resource_nodes = [r for r in self.get_useable_resources() if r.cpu_count >= cpus and r.memory >= memory]
+        if len(resource_nodes) == 0:
+            return [None]
 
-    def allocate_fill_nodes(self, cpus: int, memory: int) -> List[Optional[ResourceAllocation]]:
+        node = resource_nodes[0]
+
+        if asset_grouping == AllocationAssetGrouping.BUNDLE:
+            return [self.allocate_resource(resource_id=node.resource_id, requested_cpus=cpus, requested_memory=memory)]
+
+        allocations = []
+        mem = int(memory / cpus)
+        for i in range(cpus):
+            alloc = self.allocate_resource(resource_id=node.resource_id, requested_cpus=1, requested_memory=mem)
+            # If any individual allocation failed because assets ran out ...
+            if alloc is None:
+                # ... release anything that was allocated, and then bail
```

Comment:
Sure.  I actually used `warning` rather than `info` ... it's a bit peculiar (though not necessarily an error) if we checked earlier, saw there were enough resources on that node, but then couldn't allocate as much as needed.